regression-test/suites/load_p0/broker_load/test_broker_load_with_merge.groovy

```diff
@@ -0,0 +1,259 @@
+    }
+
+
+    // if 'enableHdfs' in regression-conf.groovy has been set to true,
+    // the test will run these case as below.
+    if (enableHdfs()) {
+        brokerName = getBrokerName()
+        hdfsUser = getHdfsUser()
+        hdfsPasswd = getHdfsPasswd()
+        def hdfs_csv_file_path = uploadToHdfs "load_p0/broker_load/broker_load_with_merge.csv"
+
+        // case1: has delete on condition and without merge
+        try {
+            sql "DROP TABLE IF EXISTS ${testTable}"
+            create_test_table.call(testTable)
+
+            def test_load_label = UUID.randomUUID().toString().replaceAll("-", "")
+            try {
+                load_from_hdfs_check_merge_type_1.call(testTable, test_load_label, hdfs_csv_file_path, "csv",
+                        brokerName, hdfsUser, hdfsPasswd)
+            }catch (Exception e) {
+                log.info(e.getMessage())
+                assertTrue(e.getMessage().contains("not support DELETE ON clause when merge type is not MERGE."))
+            }
+
+        } finally {
+            try_sql("DROP TABLE IF EXISTS ${testTable}")
+        }
+
+        // case2: has merge with delete on condition
+        try {
+            sql "DROP TABLE IF EXISTS ${testTable}"
+            create_test_table.call(testTable)
+
+            def test_load_label = UUID.randomUUID().toString().replaceAll("-", "")
+
+            try {
+                load_from_hdfs_check_merge_type_2.call(testTable, test_load_label, hdfs_csv_file_path, "csv",
+                        brokerName, hdfsUser, hdfsPasswd)
+            }catch (Exception e) {
+                log.info(e.getMessage())
+                assertTrue(e.getMessage().contains("Excepted DELETE ON clause when merge type is MERGE."))
+            }
+        } finally {
+            try_sql("DROP TABLE IF EXISTS ${testTable}")
+        }
+
+        // case3: has merge with NEGATIVEs on condition
+        try {
+            sql "DROP TABLE IF EXISTS ${testTable}"
+            create_test_table.call(testTable)
+
+            def test_load_label = UUID.randomUUID().toString().replaceAll("-", "")
+
+            try {
+                load_from_hdfs_check_merge_type_3.call(testTable, test_load_label, hdfs_csv_file_path, "csv",
+                        brokerName, hdfsUser, hdfsPasswd)
+            }catch (Exception e) {
+                log.info(e.getMessage())
+                assertTrue(e.getMessage().contains("not support MERGE or DELETE with NEGATIVE."))
+            }
+        } finally {
+            try_sql("DROP TABLE IF EXISTS ${testTable}")
+        }
+
+
+        // case4: agg table and merge type
+        try {
+            sql "DROP TABLE IF EXISTS ${testTable}"
+            create_agg_test_table.call(testTable)
+
+            def test_load_label = UUID.randomUUID().toString().replaceAll("-", "")
+
+            try {
+                load_from_hdfs_check_merge_type_4.call(testTable, test_load_label, hdfs_csv_file_path, "csv",
```

Comment:
if `load_from_hdfs_check_merge_type_4` does not throw exception,
your `assertTrue` will not be executed